packages/proto-encoder/scripts/get-proto.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+set -o errexit -o nounset -o pipefail
+command -v shellcheck >/dev/null && shellcheck "$0"
+
+PROTO_DIR="./chain-proto"
+COSMOS_SDK_VERSION="v0.47.1"
+
+OSMOSIS_COMMIT_HASH="ac133fdd73dbd27b8bc7b6d49049f5d594e771cb"
+COSMOS_COMMIT_HASH="6ed904cdb75fe08a9310e7df49eaf9c5b2b64692"
+IBC_COMMIT_HASH="10324f5bf6840892cd7bba8a40231fc52a8b1745"
+WASMD_COMMIT_HASH="4c906d5a53a255c551d6ed981a548cffe47ae9f0"
+ICS23_COMMIT_HASH="f4deb054b697458e7f0aa353c2f45a365361e895"
+
```

Comment:
For the osmosis-proto-tool I made: using ChatGPT (I don't know bash) I was actually able to extract the repo tags/versions Osmosis depends on in it's go.mod file in the 1-get-proto.sh script. By default, it gets the latest version of Osmosis on main branch, then all it's dependencies. If you want, you can supply the Osmosis commit hash as a param to get a specific version of Osmosis.

Is there a reason you used all commit hashes?

See:
https://github.com/osmosis-labs/osmosis-proto-tool/blob/0630cd5541b8d1227c2c350239d204f8bef6823d/1-get-proto.sh#L61

Reply:
Oh didn't notice that. I'll use this for cosmos, ibc and wasm versions, thanks!